File: spring_webapp1/src/main/java/com/example/spring_webapp1/domain/printing/ReceiptDocument.java
package com.example.spring_webapp1.domain.printing;

import java.util.List;

/**
 * 印字対象のレシート全体を表す値オブジェクト。
 */
public record ReceiptDocument(
        List<ReceiptLine> lines,
        String logoPath
) {
    public ReceiptDocument(List<ReceiptLine> lines) {
        this(lines, null);
    }
}

File: MauiApp1/Domain/Printing/ReceiptDocument.cs
namespace MauiApp1.Domain.Printing;

/// <summary>
/// 印字対象のレシート全体を表す値オブジェクト。
/// </summary>
public sealed record ReceiptDocument(
    IReadOnlyList<ReceiptLine> Lines,
    string? LogoPath = null
);
